src/transformers/keras_callbacks.py

```diff
@@ -334,14 +333,14 @@ def __init__(
             raise ValueError("Please supply a positive integer argument for save_steps when save_strategy == 'steps'!")
         self.save_steps = save_steps
         output_dir = Path(output_dir)
-        if hub_model_id is None:
-            hub_model_id = output_dir.absolute().name
-        if "/" not in hub_model_id:
-            hub_model_id = get_full_repo_name(hub_model_id, token=hub_token)
+
+        # Create repo and retrieve repo_id
+        self.hub_model_id = create_repo(
+            repo_id=hub_model_id or output_dir.absolute().name,
```

Comment:
Should be explicitly tested, and probably on its own line before, so we can then have everything here in the same line.